Run MP Reactive Messaging Kafka tests with Kafka client 4.0.0

Azquelt opened the pull request:
v4 of the client requires Java 11, so package a v3 client as well and use that if running on an earlier Java version.

Additional changes:
- Make the logic that packages the Kafka client into the autoFVT zip common
- Remove many unneeded dependencies from `com.ibm.ws.microprofile.reactive.messaging.kafka.secure_fat` which had been copied from another project
- Don't include transitive dependencies of the Kafka connector implementation in `com.ibm.ws.microprofile.reactive.messaging_fat`

Fixes #31227

- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

Azquelt committed:
Run RM tests with Kafka client 4.0.0

v4 of the client requires Java 11, so package a v3 client as well and
use that if running on an earlier Java version.

Additional changes:
- Make the logic that packages the Kafka client into the autoFVT zip
  common
- Remove many unneeded dependencies from
  com.ibm.ws.microprofile.reactive.messaging.kafka.secure_fat which had
  been copied from another project
- Don't include transitive dependencies of the Kafka connector
  implementation in com.ibm.ws.microprofile.reactive.messaging_fat

Azquelt committed:
Fix Reactive Streams TCK within RM FAT

We run the Reactive Streams TCK against our Kafka consumer and producer.
However, it wasn't working because we'd used the Junit BeforeClass
annotation instead of the TestNG annotation and the results weren't
being put into the correct directory for the FAT framework to pick up
and report the failure.

Azquelt commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=431f9f92-ebd2-4b43-b182-d74fe07d6c42) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

Azquelt commented:
SOE build: https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=236e13ee-eed2-4b2f-a254-cc721adb2c8b

abutch3r reviewed (dismissed):
LGTM to me.

And great we fixed somme of the broken tests

Will need to verify the tests once the PB has completed to make sure we got the desired behavior across the 1.0 FATs.

Azquelt commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=17136865-5634-473b-b55b-b2a73e8fd2cb) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

Azquelt commented:
SOE build: https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=e0a01156-ed99-4fc1-9bcb-c5723eef0ed0

Only change in the last push was to update copyright headers.

Azquelt committed:
Add permissions required by Kafka Clients 4

Ironically, Kafka clients 4 uses reflection to call the SecurityManager
so that it can cope with classes being missing, which means it needs
additional reflection permissions.

Azquelt committed:
Fix RM FAT server feature list

Include mpConfig so that we run with the expected mpConfig version when
doing repeats, otherwise we end up using mpConfig-3.0 instead of
mpConfig-3.1 in the MP61 configuration.

Azquelt commented:
Annoyingly, running locally turned up some security manager issues. I have to pause work on this, but we still need to:
- Update SharedLibRxMessagingServer/server.xml to apply permissions to the 4.0 Kafka client jar
- Run the test buckets locally in full mode
  - There are checks for SecurityManager and that feature repeats are working correctly that don't run in the build

Azquelt commented:
!build
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=b4710e0d-0996-4a5c-9cad-4e30679c576e) - :x: completed with errors/failures.)
spawn.fullfat.buckets=com.ibm.ws.microprofile.reactive.messaging_fat,com.ibm.ws.microprofile.reactive.messaging.kafka.secure_fat,io.openliberty.microprofile.reactive.messaging.internal_fat

_Note: Target locations of links might be accessible only to IBM employees._

Azquelt commented:
SOE build: https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=0c9e4adb-b216-4ce0-b05a-416b23ce8fad

Azquelt commented:
Forgot to update the copyright headers

!build spawn.fullfat.buckets=com.ibm.ws.microprofile.reactive.messaging_fat,com.ibm.ws.microprofile.reactive.messaging.kafka.secure_fat,io.openliberty.microprofile.reactive.messaging.internal_fat ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=4de167be-8b72-43c7-a234-15181cc92f15) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

Azquelt commented:
SOE build: https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=7b82f333-2ea2-4b56-bc86-e33fef089b31

Azquelt commented:
Failures in the build do not appear to be related to this change. SOE build is good.

abutch3r reviewed (approved):
LGTM, Can confirm SOE and PB was clean.

Explanation for why mpConfig being added to make sure repeats update the version of it correctly
